client/src/app/pages/environment-list/environment-list.component.ts

```diff
@@ -26,6 +26,17 @@ import {DeploymentDTO} from '@app/core/modules/openapi/model/deployment-dto';
 export class EnvironmentListComponent {
   environmentService = inject(EnvironmentControllerService);
   deploymentService = inject(DeploymentControllerService);
+  route = inject(ActivatedRoute);
```

Comment:
nitpick: Instead of manually getting the params you can also just add an input with the params name, angular will assign the route params value automatically:
`projectId = input.required({ transform: numberAttribute });`